addon/resolve-attribute-util.js

```diff
@@ -12,19 +12,20 @@ import {
 } from './utils/resolve';
 
 function resolveReference(store, reference) {
+  let { id } = reference;
   if (reference.type === null) {
     // for schemas with a global id-space but multiple types, schemas may
     // report a type of null
-    let internalModel = store._globalM3Cache[reference.id];
+    let internalModel = store._globalM3Cache[id];
     return internalModel ? internalModel.getRecord() : null;
   } else {
     // respect the user schema's type if provided
-    return store.peekRecord(reference.type, reference.id);
+    return id !== null && id !== undefined ? store.peekRecord(reference.type, reference.id) : null;
```

Comment:
when do we have this case of having a reference but not a record?

Reply:
The algorithm of check for ref then nested model then raw value isn’t great; we should change it to one hook on the user schema where they pick (this saves us from some silly dancing between computeAttributeReference and computeNestedModel).

Cases where you want to explicitly say i have a reference with no record are mainly where you have a difference between the object properties the user accesses and the underlying keys in the payload